src/figuresClasses.ts

```diff
@@ -1,19 +1,109 @@
 export interface Figure {
+  shape: string,
+  color: string,
+  getArea(): number,
+}
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
-}
+export class Triangle implements Figure {
+  public color: Color;
+
+  public shape: Shape;
+
+  public a: number;
+
+  public b: number;
+
+  public c: number;
 
-export class Circle {
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.shape = Shape.Triangle;
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
check checklist again (I updated it) and remove it

Reply:
![image](https://user-images.githubusercontent.com/55781950/234829253-40dea0da-618d-4389-8bc1-aeddd888058d.png)
Linter disagrees

Reply:
This is because you should also delete it's declaration above the constructor

Reply:
![image](https://user-images.githubusercontent.com/55781950/236172131-069a50ea-02e5-4e41-9f22-97efc813fab9.png)

Reply:
The linter does not let me do it in any way that does not have:

- A public declaration above the constructor
- A parameter reference
- A this.color

HOWEVER
It seems to be working fine with a, b, and c

Reply:
HOWEVER HOWEVER
Doing that does not allow the getArea function to read a, b, or c

Reply:
![image](https://user-images.githubusercontent.com/55781950/236173117-2f1286bf-532f-4b8e-8c64-6d04b5a69634.png)
as seen here